python_modules/dagster/dagster/_core/definitions/partitions/definition/time_window.py

```diff
@@ -95,6 +100,7 @@ class TimeWindowPartitionsDefinition(PartitionsDefinition, IHaveNew):
     fmt: PublicAttr[str]
     end_offset: PublicAttr[int]
     cron_schedule: PublicAttr[str]
+    exclusions: PublicAttr[Optional[set[Union[str, TimestampWithTimezone]]]]
```

Comment:
somewhat concerned that the fact that this is a set will mess up snapshots in some way because the representation will be unordered? e.g. if you have exclusions then the hashes for your code location will not be stable because it changes every time it's serialized? I could definitely be overthinking this, not sure exactly how we end up serializing sets, but worth adding a test where this is set and we compute snapshot hashes

Reply:
hmm, rather than figuring out a sort order for the str/timestamp items, I'm going to switch this to a sequence